### PR TITLE
Update expanded registers

### LINKED ISSUE #20: calling convention for sN registers of REX
Commit 7f66dbe updates the documentation of the [Expanded Registers](https://github.com/ETC-A/etca-spec/blob/main/extensions/expanded-registers) extension, to the current names for the added registers. I think its "who saves it" column may need an update to match.

Formerly, the registers r13-r15 had aliases a8, t0, and t1, and correspondingly were listed as call**er**-saved. Register aliases s2-s4 were on registers r24-r26, and were call**ee**-saved. Now r13-r15 _are_ s2-s4, but they've kept their call**er**-saved recommendation, which seems inconsistent with the caption text that "sN registers are registers that must be saved before re-using."

Should they be updated to call**ee**-saved, like the other sN registers? Or is it intended that these registers keep the call**er**-saved status?

### PR DESCRIPTION
**Pull request by Ezra**
Resolves #20 and makes some stylistic edits to the surrounding text.

### CODE REVIEW

**Review by EnderShadow — CHANGES_REQUESTED** [review]
I prefer that we stick with "caller saved" and "callee saved" since that has a technical meaning whereas "call-clobbered" and "call-preserved" don't have the same meaning.

**Comment by JKTKops** [comment]
Indeed, caller saved and callee saved are technical terms. The confusion is unfortunate but they are standard now.

File: extensions/stack-and-functions/README.md [hunk 1]
```diff
@@ -9,24 +9,24 @@
...
-- tN registers are temporary registers which a function can use however it wants.
+- vN registers are the first N return values from a function call. Additional return values should be pushed to the stack.
+- sN registers are registers that must be saved before, and restored after, use.
+- tN registers (see the [Expanded Registers](../expanded-registers/README.md) extension) are temporary registers which a function can use however it wants.
 - bp is the base pointer register and stores the address of the bottom of the stack for this function.
-- sp is the stack pointer register and stores the address of the top of the stack for this function. This is hardcoded into the push and pop instructions
-- ln is the link register and stores the address that this function should return to. This is hard coded into the call and return instructions
+- sp is the stack pointer register and stores the address of the top of the stack for this function. This is hardcoded into the push and pop instructions.
+- ln is the link register and stores the address that this function should return to. This is hard coded into the call and return instructions.
```
Comment (JKTKops) [inline extensions/stack-and-functions/README.md]:
```suggestion
- ln is the link register and stores the address that this function should return to. This is hardcoded into the call and return instructions.
```

File: extensions/stack-and-functions/README.md [hunk 2]
```diff
@@ -9,24 +9,24 @@
...
+| Register | Alias     | Calling Convention |
+|----------|-----------|----------------|
+| `r0`     | `a0`/`v0` | call-clobbered |
+| `r1`     | `a1`/`v1` | call-clobbered |
+| `r2`     | `a2`      | call-clobbered |
+| `r3`     | `s0`      | call-preserved |
+| `r4`     | `s1`      | call-preserved |
+| `r5`     | `bp`      | call-preserved |
+| `r6`     | `sp`      | call-preserved |
+| `r7`     | `ln`      | call-clobbered |
```
Comment (JKTKops) [inline extensions/stack-and-functions/README.md]:
```suggestion
| Register | Alias     | Calling Convention |
|----------|-----------|----------------|
| `r0`     | `a0`/`v0` | caller saved |
| `r1`     | `a1`/`v1` | caller saved |
| `r2`     | `a2`      | caller saved |
| `r3`     | `s0`      | callee saved |
| `r4`     | `s1`      | callee saved |
| `r5`     | `bp`      | callee saved |
| `r6`     | `sp`      | callee saved |
| `r7`     | `ln`      | caller saved |
```
Capitalization doesn't matter much to me. They were capitalized before but leaving them lowercase is fine.

File: extensions/expanded-registers/README.md [hunk 3]
```diff
@@ -40,3 +23,22 @@ When used with an instruction that uses an `SIB.X` register, `REX.X` provides a
...
+| Register | Alias | Calling Convention |
+|----------|-------|----------------|
+| `r8`     | `t0`  | call-clobbered |
+| `r9`     | `t1`  | call-clobbered |
+| `r10`    | `t2`  | call-clobbered |
+| `r11`    | `t3`  | call-clobbered |
+| `r12`    | `t4`  | call-clobbered |
+| `r13`    | `s2`  | call-preserved |
+| `r14`    | `s3`  | call-preserved |
+| `r15`    | `s4`  | call-preserved |
```
Comment (JKTKops) [inline extensions/expanded-registers/README.md:39]: Same as suggested change below; use technical terms.

File: extensions/expanded-registers/README.md [hunk 4]
```diff
@@ -7,31 +7,14 @@
...
+This extension adds an instruction prefix byte expanding the number of available registers from 8 registers to 16, and suggests an extension of the [Stack & Functions](../stack-and-functions/readme.md) extension's calling conventions to the added registers.
 
 # Added Instructions
 
 The following 1-byte instruction prefix is now defined.
 
-| `REX`   | | | | |
-|:-------------|---|----|----|----|
+| `REX`  |     |     |     |     |
+|:-------|-----|-----|-----|-----|
```
Comment (JKTKops) [inline extensions/expanded-registers/README.md:17]: Cleaning these up is fine, but remember that markdown doesn't actually care about the spacing and spacing things "correctly" gets messed up if the table is ever modified. This has come up in our tables before.

**Review by JKTKops — CHANGES_REQUESTED** [review]

**Comment by EnderShadow** [comment]
As this has been idle for a bit, I'm closing it and putting the changes in #24

**Pull request closed without merging**